### ISSUE TITLE
AWS Lambda Requires Two `waypoint up` runs

### ISSUE THREAD

**Issue by onlydole** [hashicorp/waypoint#1271]
**Describe the bug**

After running `waypoint up` for an AWS Lambda function, I get an `InvalidParameterValueException` but when I run `waypoint up` a second time, it works without any issue.

**Steps to Reproduce**

1. Use the example from the link below to instantiate  your workspace
2. Install the Waypoint server
3. Run `waypoint up`
4. Get the following message in your CLI

```bash
Docker image pushed:
xxx.dkr.ecr.us-west-2.amazonaws.com/learn-waypoint-lambda:howdy-waypoint-lambda

» Deploying...
✓ Connecting to AWS
❌ Creating new Lambda function
! InvalidParameterValueException: The role defined for the function cannot be
  assumed by Lambda.
  {
    RespMetadata: {
      StatusCode: 400,
      RequestID: "xxx"
    },
    Message_: "The role defined for the function cannot be assumed by Lambda.",
    Type: "User"
  }
```

[Example Link](https://github.com/hashicorp/waypoint-examples/tree/f1afc14796e7ceb9a5479ee4dacf9192e15c67dc/aws/lambda/ruby) 

**Expected behavior**

I would like to be able to run `waypoint up` and the Lambda Role to work without having to run this again. 

**Waypoint Platform Versions**

* Waypoint CLI Version: Waypoint v0.2.0-939-gc3f241b6 (c3f241b6)
* Waypoint Server Platform and Version: Docker, Waypoint v0.2.0-939-gc3f241b6

**Comment by evanphx** [comment]
This was fixed directly prior to 0.3.0.

**Comment by onlydole** [comment]
Yessss! Thank you, @evanphx!
